cli/kaos_cli/services/tests/test_state_service.py

```diff
@@ -0,0 +1,228 @@
+import os
+import time
+from parameterized import parameterized
+from configobj import ConfigObj
+
+from unittest import TestCase
+
+from kaos_cli.services.state_service import StateService
+from kaos_cli.constants import CONFIG_PATH, KAOS_STATE_DIR
+
+
+def state_service_test_inputs():
+    return [
+        (
+            'context1', 'param1'
+        ),
+        (
+            'context2', 'param2'
+        ),
+        (
+            '', 'param3'
+        ),
+        (
+            'context4', ''
+        ),
+        (
+            '', ''
+        ),
+        (
+            {}, 'param6'
+        ),
+        (
+            'context7', {}
+        ),
+        (
+            {}, {}
+        ),
+        (
+            '', {}
+        ),
+        (
+            {}, ''
+        ),
+        (
+            None, 'param11'
+        ),
+        (
+            'context11', None
+        ),
+        (
+            None, None
+        ),
+        (
+            {}, None
+        ),
+        (
+            None, {}
+        ),
+        (
+            '', None
+        ),
+        (
+            None, ''
+        ),
+        (
+            1, 'param18'
+        ),
+        (
+            'context', 2
+        ),
+        (
+            3, {}
+        ),
+        (
+            {}, 4
+        ),
+        (
+            5, ''
+        ),
+        (
+            '', 6
+        ),
+        (
+            7, None
+        ),
+        (
+            None, 8
+        ),
+
+    ]
+
+
+class TestStateService(TestCase):
+
+    # Arrange
+    TestCase.state_service = StateService()
+    TestCase.switch = False
```

Comment:
This is an anti-pattern. Please avoid maintaining state between unit tests as you won't be able to isolated the causes and effects that are testing for properly. Instead, please create an instance of the `StateService` in each unit test and carry out the mutations you're testing for inside that test. The test should have no side-effect, neither local nor global, neither inside nor outside the method or the test case.

Reply:
Separate instantiation of ```StateService``` now occurs for each unit test